### PR TITLE
feat(clients/go): add credentials support in Go client

### LINKED ISSUE #2840: Add JWT in the Go client
### Objective
The goal of this issue is to add a JWT to gRPC calls such that a server can validate it and ensure that the client is authorized to access a particular endpoint. 

### Expected Behavior
Same as the Java client #2839 

Blocked by #2838

### PR DESCRIPTION
**Pull request by miguelpires**
## Description
Added credentials support to the Go client. 

## Related issues

closes #2840 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/go/zbc/credentialsProvider.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"log"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// CredentialsProvider is responsible for adding credentials to each gRPC call's headers.
+type CredentialsProvider interface {
+	ApplyCredentials(*map[string]string)
```
Comment (npepinpe) [inline clients/go/zbc/credentialsProvider.go]: Can we document was the input is?

**Review by npepinpe — CHANGES_REQUESTED** [review]
Looks good, just add the changes we discussed

**Comment by MiguelPires** [comment]
Thanks for the feedback @npepinpe. I made the adjustments we mentioned

**Review by npepinpe — APPROVED** [review]
:+1:

**Comment by MiguelPires** [comment]
Bors retry

**Pull request merged**